### DISCUSSION TITLE
I am update dreamlight_led_strip.yaml - adding scenes selector / values

### DISCUSSION THREAD

**Discussion by Levanterman** [make-all/tuya-local#2324] [Q&A]
This form is for reporting bugs.  If you have encountered a bug, you should
generally have some log messages or device diagnostics to show something is
going wrong.  If your problem is more of a question or a request for help,
then [Discussions](https://github.com/make-all/tuya-local/discussions) is
probably a more suitable venue.

**Describe the bug**
I am trying to modify the existing dreamlight_led_strip.yaml   The device adds to Tuya Local no problem, but you cant change the 'scenes' (only from the app) 
[https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/README.md](https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/README.md)
<img width="953" alt="image" src="https://github.com/user-attachments/assets/b40acd79-b831-4c43-8b36-88e0e819077e">

**To Reproduce**
Steps to reproduce the behavior including the device the issue was observed with:
This is the device info from Tuya:
```
{
  "result": {
    "model": "{\"modelId\":\"000003xq3l\",\"services\":[{\"actions\":[],\"code\":\"\",\"description\":\"\",\"events\":[],\"name\":\"默认服务\",\"properties\":[{\"abilityId\":20,\"accessMode\":\"rw\",\"code\":\"switch_led\",\"description\":\"\",\"extensions\":{\"iconName\":\"icon-dp_power\",\"attribute\":\"646\"},\"name\":\"开关\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":21,\"accessMode\":\"rw\",\"code\":\"work_mode\",\"description\":\"\",\"extensions\":{\"iconName\":\"icon-dp_mode\",\"attribute\":\"640\"},\"name\":\"模式\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"white\",\"colour\",\"scene\",\"music\"]}},{\"abilityId\":24,\"accessMode\":\"rw\",\"code\":\"colour_data\",\"description\":\"类型：字符；\\nValue: 000011112222；\\n0000：H（色度：0-360，0X0000-0X0168）；\\n1111：S (饱和：0-1000, 0X0000-0X03E8）；\\n2222：V (明度：0-1000，0X0000-0X03E8)\",\"extensions\":{\"iconName\":\"icon-yanse\",\"attribute\":\"512\"},\"name\":\"彩光\",\"typeSpec\":{\"type\":\"string\",\"maxlen\":255}},{\"abilityId\":26,\"accessMode\":\"rw\",\"code\":\"countdown\",\"description\":\"\",\"extensions\":{\"iconName\":\"icon-dp_time2\",\"attribute\":\"640\"},\"name\":\"倒计时剩余时间\",\"typeSpec\":{\"type\":\"value\",\"max\":86400,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"s\"}},{\"abilityId\":27,\"accessMode\":\"wr\",\"code\":\"music_data\",\"description\":\"类型：字符串；\\nValue: 011112222333344445555；\\n0：   变化方式，0表示直接输出，1表示渐变；\\n1111：H（色度：0-360，0X0000-0X0168）；\\n2222：S (饱和：0-1000, 0X0000-0X03E8）；\\n3333：V (明度：0-1000，0X0000-0X03E8）；\\n4444：白光亮度（0-1000）；\\n5555：色温值（0-1000）\",\"extensions\":{\"iconName\":\"icon-dp_mic\",\"attribute\":\"640\"},\"name\":\"音乐灯\",\"typeSpec\":{\"type\":\"string\",\"maxlen\":255}},{\"abilityId\":46,\"accessMode\":\"ro\",\"code\":\"light_length\",\"description\":\"幻彩灯带在UI面板上展示的总长度\",\"extensions\":{\"iconName\":\"icon-setting\",\"attribute\":\"512\"},\"name\":\"灯带长度\",\"typeSpec\":{\"type\":\"value\",\"max\":10000,\"min\":1,\"scale\":0,\"step\":1,\"unit\":\"cm\"}},{\"abilityId\":47,\"accessMode\":\"ro\",\"code\":\"light_pixel\",\"description\":\"幻彩灯带驱动芯片总数\",\"extensions\":{\"iconName\":\"icon-setting\",\"attribute\":\"512\"},\"name\":\"灯带点数\",\"typeSpec\":{\"type\":\"value\",\"max\":1024,\"min\":1,\"scale\":0,\"step\":1}},{\"abilityId\":51,\"accessMode\":\"rw\",\"code\":\"dreamlight_scene_mode\",\"description\":\"专门用于幻彩灯带场景\",\"extensions\":{\"iconName\":\"icon-dp_mode\",\"attribute\":\"640\"},\"name\":\"幻彩情景\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":52,\"accessMode\":\"rw\",\"code\":\"dreamlightmic_music_data\",\"description\":\"专门用于幻彩灯带本地音乐律动\",\"extensions\":{\"iconName\":\"icon-dp_mic\",\"attribute\":\"128\"},\"name\":\"幻彩本地音乐律动\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":53,\"accessMode\":\"rw\",\"code\":\"lightpixel_number_set\",\"description\":\"幻彩灯带裁剪之后，对长度重新进行设置\",\"extensions\":{\"iconName\":\"icon-gongneng\",\"attribute\":\"640\"},\"name\":\"点数/长度设置\",\"typeSpec\":{\"type\":\"value\",\"max\":1000,\"min\":1,\"scale\":1,\"step\":1}},{\"abilityId\":61,\"accessMode\":\"rw\",\"code\":\"paint_colour_data\",\"description\":\"专门用于幻彩灯带的调光调色\",\"extensions\":{\"iconName\":\"icon-dp_circle\",\"attribute\":\"512\"},\"name\":\"涂抹调色\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}}]}]}"
  },
  "success": true,
  "t": 1726934953897,
  "tid": "d83a4223783311ef88b02ac421ecbce6"
}
```
**Expected behavior**

**Additional context**
![image](https://github.com/user-attachments/assets/f26b4de0-bfc8-4d9f-a8ca-6fe6cda56b31)

The attribute I am trying to alter is no.51 but I dont know how to map a type : "raw" into the yaml, tuya documentation says its binary but i am not sure how to send different binary values to each scene
Please can you give me a hint, thanks. 

If the bug involves a device, then please include device diagnostics from
Settings / Devices & Services / Tuya Local (your device) / "1 device"

Please check if there are messages from Tuya Local in the Home
Assistant log (Settings / System / Logs), and include those too.

These are the some of the scenes which can be selected in the Tuya App.
![image](https://github.com/user-attachments/assets/eaa7adda-a692-4a5c-a03e-5e845ad156ad)

Looks like the binary value is sent as a string, but need to concatenate other values into the correct format 
<img width="1091" alt="image" src="https://github.com/user-attachments/assets/ae728fa8-57f8-4e83-84d6-87b39d609477">
[https://developer.tuya.com/en/docs/iot/11?id=K9tp116dmo6br](url)

i found the values in Tuya , i can change the scenes in the app and this value changes.
![image](https://github.com/user-attachments/assets/d29184aa-2cfd-4abf-8a96-47586ffce668)

I saw how to map the static values in another yaml (smart_ked_strip.yaml) but these scenes dont work on this model

![image](https://github.com/user-attachments/assets/82560ecd-bb76-4552-85f4-cdd89455d1ab)

aha got it working and its accepting the scene values.. 
I will need to work my way though all the scenes , query the values from Tuya and then update the Yaml. 
This will be a Pull request?
![image](https://github.com/user-attachments/assets/8111267c-b96b-4c12-95f9-58717d9dca75)

just wondering if I am missing something , as I know the scene values are correct, thanks

**Comment by make-all** [comment]
This is the correct way to do it.

I am not sure whether you still have any questions here, as the original request seems to have been written as you were discovering things, and it seems you solved all the questions you had by yourself.

**Reply by Levanterman** [reply]
Ha yes, it was exactly that, sorry. Started off with the problem, but found the solution eventually. Once I have finished the yaml i'll do a pull request for this different model of Tuya LED meon Rope Light with scenes. Thanks.

**Reply by thestumonkey** [reply]
Did you manage to figure out how to set the individual LED's in the chain?

**Reply by Levanterman** [reply]
> Did you manage to figure out how to set the individual LED's in the chain?

Hi, I figured out how to set the in built effects, which works great (I just need to submit the code into a future release but its ready to go). As for individuial LED's , I did not try this, the scenes/effects are pretty good and they control each led , but I dont know of a custom way to control each led. 
n.

**Reply by make-all** [reply]
There is an example in enbrighten_cafe_stringlights.yaml that may be similar.
